src/content/docs/apm/agents/php-agent/php-agent-api/newrelic_set_error_group_callback.mdx

```diff
@@ -0,0 +1,238 @@
+---
+title: newrelic_set_error_group_callback (PHP agent API)
+type: apiDoc
+shortDescription: This method allows a user to define a custom callback function to generate error group names.
+tags:
+  - Agents
+  - PHP agent
+  - PHP agent API
+metaDescription: 'Use this API to register a callback which returns a custom error group name.'
+redirects:
+  - /docs/agents/php-agent/php-agent-api/newrelic_set_error_group_callback
+---
+
+Group errors with your own custom-defined fingerprinting callback function.
+
+## Syntax
+
+```php
+newrelic_set_error_group_callback($callback)
+```
+
+## Requirements
+
+Agent version 10.12 or higher.
+
+## Description
+
+This API allows a user to register a custom callback function with the PHP Agent that will be called when the application encounters an error.
+
+The callback will provided two PHP arrays from the Agent- one containing transaction data, another containing error data.
+
+Using the provided context information, the callback must return a non-empty string based on user-defined logic that will serve as the error group name displayed in the Errors Inbox UI.
+
+<Callout variant="important">
+  - When there are multiple calls to this function in a single transaction, the PHP agent retains the callback from the last call only.
+
+  - Callbacks are registered on a per-request basis. This API should be called in a code path that is guaranteed to be executed for each request, or the callback will not be invoked.
+
+  - It is highly recommended to keep your callback function as minimal as possible. Cpu-intensive callbacks (such as database calls) or other complex logic will result in performance impacts to your application.
+</Callout>
+
+## Parameters
+
+This API accepts a single function-type callback as an argument. The provided callback must accept 2 parameters.
+
+<table>
+  <thead>
+    <tr>
+      <th width="25%">
+        API Parameter
+      </th>
+
+      <th>
+        Description
+      </th>
+    </tr>
+  </thead>
+
+  <tbody>
+    <tr>
+      <td>
+        `$callback` (_function_)
+      </td>
+
+      <td>
+        Required. Provide a callback function that will be registered with the PHP Agent.
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+<table>
+  <thead>
+    <tr>
+      <th width="25%">
+        Callback Parameters
+      </th>
+
+      <th>
+        Description
+      </th>
+    </tr>
+  </thead>
+
+  <tbody>
+    <tr>
+      <td>
+        `$txndata` (_array_)
+      </td>
+
+      <td>
+        Required. An array of transaction data provided to your callback by the PHP Agent.
+      </td>
+    </tr>
+    <tr>
+     <td>
+       `$errdata` (_array_)
```

Comment:
```suggestion
       `$error_data` (_array_)
```

Reply:
Applied changes in bb07f23bde149d3f160cd482fb45db3236b60519